File: CS/Retain_Img_Src/Form1.cs
using System;
using System.Windows.Forms;
#region #usings
using DevExpress.XtraRichEdit.Services;
using DevExpress.Office.Utils;
#endregion #usings

namespace Retain_Img_Src
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            richEditControl1.LoadDocument("test.htm");
            richEditControl1.ContentChanged+=new EventHandler(richEditControl1_ContentChanged);
        }

        private void richEditControl1_ContentChanged(object sender, EventArgs e)
        {
            memoEdit1.Text = richEditControl1.Document.GetHtmlText(richEditControl1.Document.Range, new CustomUriProvider());
        }
        #region #documentloaded
        private void richEditControl1_DocumentLoaded(object sender, EventArgs e)
        {
            IUriProviderService service = richEditControl1.GetService<IUriProviderService>();
            if (service != null) {
                service.RegisterProvider(new CustomUriProvider());
            }
        }
        #endregion #documentloaded
    }
    #region #customuriprovider
    public class CustomUriProvider : IUriProvider
    {
        #region IUriProvider Members
        public string CreateCssUri(string rootUri, string styleText, string relativeUri)
        {
            return String.Empty;
        }

        public string CreateImageUri(string rootUri, OfficeImage image, string relativeUri)
        {
            return image.Uri;
        }
        #endregion
    }
    #endregion #customuriprovider
}

File: VB/Retain_Img_Src/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
#Region "#usings"
Imports DevExpress.XtraRichEdit.Services
Imports DevExpress.Office.Utils
#End Region ' #usings

Namespace Retain_Img_Src
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			richEditControl1.LoadDocument("test.htm")
			AddHandler richEditControl1.ContentChanged, AddressOf richEditControl1_ContentChanged
		End Sub

		Private Sub richEditControl1_ContentChanged(ByVal sender As Object, ByVal e As EventArgs)
			memoEdit1.Text = richEditControl1.Document.GetHtmlText(richEditControl1.Document.Range, New CustomUriProvider())
		End Sub
		#Region "#documentloaded"
		Private Sub richEditControl1_DocumentLoaded(ByVal sender As Object, ByVal e As EventArgs) Handles richEditControl1.DocumentLoaded
			Dim service As IUriProviderService = richEditControl1.GetService(Of IUriProviderService)()
			If service IsNot Nothing Then
				service.RegisterProvider(New CustomUriProvider())
			End If
		End Sub
		#End Region ' #documentloaded
	End Class
	#Region "#customuriprovider"
	Public Class CustomUriProvider
		Implements IUriProvider
		#Region "IUriProvider Members"
		Public Function CreateCssUri(ByVal rootUri As String, ByVal styleText As String, ByVal relativeUri As String) As String Implements  IUriProvider.CreateCssUri
			Return String.Empty
		End Function

		Public Function CreateImageUri(ByVal rootUri As String, ByVal image As OfficeImage, ByVal relativeUri As String) As String Implements  IUriProvider.CreateImageUri
			Return image.Uri
		End Function
		#End Region
	End Class
	#End Region ' #customuriprovider
End Namespace